GVFS/GVFS.Common/InstallerPreRunChecker.cs

```diff
@@ -32,13 +34,21 @@ public virtual bool TryRunPreUpgradeChecks(out string consoleError)
                 {
                     consoleError = $"{GVFSConstants.UpgradeVerbMessages.GVFSUpgrade} is not supported in unattended mode";
                     this.tracer.RelatedError($"{nameof(this.TryRunPreUpgradeChecks)}: {consoleError}");
-                    return false;
+
+                    if (!this.dryRun)

```

Comment:
Just so I understand correctly `dryRun` will continue to run even in error conditions?  Is this so developers can see all the possible errors that might be hit when the upgrade is ran?

Reply:
Like I said below, I think the only distinction between dry run and real installer is that dry run should not run the installers. I think everything else should still error out so we have a good understanding of potential reliability issues.